### ISSUE TITLE
worker skips half the field

### ISSUE THREAD

**Issue by Commissar0617** [Courseplay/courseplay#6652]
![20210109185711_1](https://user-images.githubusercontent.com/5993529/104112882-8a70ae80-52b9-11eb-98ca-2c83c564b5c0.jpg)
![20210109185717_1](https://user-images.githubusercontent.com/5993529/104112886-8e9ccc00-52b9-11eb-8a26-7e30d5f7aa4b.jpg)

so, im using a 2 implement course..... one of the workers decided it doesn't want to be behind, and skips like half the field. using up/down w/ 3 headlands. 

https://mega.nz/file/fK4zEIZA#Y0QbucmpfOJmo6iyyrnff_-4sQvObK8VNXbgrN14948
log too big for github

**Comment by Commissar0617** [comment]
![20210109231347_1](https://user-images.githubusercontent.com/5993529/104115045-9c118080-52d0-11eb-9ce3-c82161dbdf68.jpg)
![20210109231408_1](https://user-images.githubusercontent.com/5993529/104115046-9caa1700-52d0-11eb-8a0f-365f958e5388.jpg)
![20210109231418_1](https://user-images.githubusercontent.com/5993529/104115047-9d42ad80-52d0-11eb-8d97-7f262e7c2d8c.jpg)
[log.txt](https://github.com/Courseplay/courseplay/files/5792272/log.txt)

more logging for ya.

**Comment by Commissar0617** [comment]
yeah, this seems to be happening with all seeders and fieldwork implements. have not noticed it with a combine

**Comment by pvaiko** [comment]
Sorry, it is probably obvious to you but I have no idea what _**exactly**_ is happening when it "skips half the field".

**Comment by Commissar0617** [comment]
Sorry man, so basically what happens is that in seeding or fieldwork, after doing one set of lanes, the worker then skips a number of lanes and continues to work. 

It just drives down the field and starts there, and runs to the end.

**Comment by pvaiko** [comment]
Show me your course generator screen for this course.

**Comment by pvaiko** [comment]
And a screenshot of the course, with the lanes being skipped marked.

**Comment by Commissar0617** [comment]
![20210111223146_1](https://user-images.githubusercontent.com/5993529/104390012-5936ef00-5502-11eb-8623-6e4521fd789f.jpg)


this was a screenshot from yesterday that should give you some idea. ill get you the other screens you wanted still. it's happened on multiple fields with multiple tractors and implements.

**Comment by Commissar0617** [comment]
for some reason, it didn't do it tonight. before, it did it on fields 90, 68, 27, and 11 on landdownunda

**Comment by pvaiko** [comment]
It is because of that insanely long rig you have there. This happens after it finishes a turn and then is looking for the next waypoint in front of it to continue on the up/down row. I'd really need to see the course in that area, for example in your second log, it was the turn starting at waypoint 1287.

**Comment by Commissar0617** [comment]
I mean, yeah, the planter is pretty insanely long.  I had to drop the  spreader off the end of the aircart for one field because i couldn't get enough headland in, but that's just an inherent problem, i wouldn't expect any sorta fix for.  


The disks i don't think are that extraordinary. My db120 is about as long unfolded. I did 2 headlands, with two implements. I believe it was the 2nd turn on the up/down in the 1st screenshot. If that helps. 

Ill see if i cant reproduce it after harvest.
